src/error.ts

```diff
@@ -121,6 +121,13 @@ export function createError<DataT = unknown>(
     err.statusCode = sanitizeStatusCode(input.statusCode, err.statusCode);
   } else if (input.status) {
     err.statusCode = sanitizeStatusCode(input.status, err.statusCode);
+  } else if (
+    typeof input.cause === "object" &&
+    input.cause &&
+    "statusCode" in input.cause &&
+    typeof input.cause.statusCode === "number"
+  ) {
+    err.statusCode = sanitizeStatusCode(input.cause.statusCode, err.statusCode);
```

Comment:
Should I also try and detect `status` from `error.cause.status` since this is also checked on the top-level properties, or should I just check `statusCode` here?